### PR TITLE
nix-prefetch: 0.3.0 -> 0.3.1

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change
Regular update and did some cleanups to make things a little neater. We
patch in the patch phase, install in the install phase and so on.

Also use some of the helpers we have available for shell completion and
man pages.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by msteen — CHANGES_REQUESTED** [review]
Thank you for making the PR! I planned to make a PR for 4.0, but it's development is on hold for the moment due to lack of time. The only request I have is whether you could add/change:
```nix
{
  src = fetchFromGitHub {
    owner = "msteen";
    repo = "nix-prefetch";
    rev = version;
    sha256 = "1bjgs52x03wydh51zl4kn6j52qwhkqy3z6wl7mvq0g31lsg9gdx4";
    extraPostFetch = ''
      echo "SOURCE_DATE_EPOCH=$(stat -c %Y $out)" > $out/.env
    '';
  };

  configurePhase = ''
    set -a
    . .env
    set +a
  '';
}
```

Such that the date is actually incorporated properly in the manpage rather than having 01/01/1970. Please correct me if it can be done better.

File: pkgs/tools/package-management/nix-prefetch/default.nix
Comment (peterhoeg) [reply pkgs/tools/package-management/nix-prefetch/default.nix:65]: Yes

File: pkgs/tools/package-management/nix-prefetch/default.nix
Comment (peterhoeg) [reply pkgs/tools/package-management/nix-prefetch/default.nix:3]: I have `nixpkgs-fmt` in a save-hook so all formatting is due to that.

**Comment by peterhoeg** [comment]
> Such that the date is actually incorporated properly in the manpage rather than having 01/01/1970. Please correct me if it can be done better.

Storing just the timestamp, we don't have to do the `set -a` dance but personally I wouldn't care about the date but that's of course your call. It's been fixed.

File: pkgs/tools/package-management/nix-prefetch/default.nix
Comment (msteen) [inline pkgs/tools/package-management/nix-prefetch/default.nix:50]: Not sure if Nix already does this for us, but do we need to add `--utc` to make it more pure?

**Comment by peterhoeg** [comment]
I'll squash this on merge.

**Review by msteen — APPROVED** [review]
I do not see any problems with it anymore, only nice improvements, and I learned some new stuff with this PR, so great job!

**Pull request merged**
